pkg/aws/cloudwatch/histograms/test_cases_test.go

```diff
@@ -0,0 +1,385 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package histograms // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/aws/cloudwatch/histograms"
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHistogramFeasibility(t *testing.T) {
+	testCases := TestCases()
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			feasible, reason := checkFeasibility(tc.Input)
+			assert.True(t, feasible, reason)
+
+			// check that the test case percentile ranges are valid
+			for percentile, expectedRange := range tc.Expected.PercentileRanges {
+				calculatedLow, calculatedHigh := calculatePercentileRange(tc.Input, percentile)
+				assert.Equal(t, expectedRange.Low, calculatedLow, "calculated low does not match expected low for percentile %v", percentile)
+				assert.Equal(t, expectedRange.High, calculatedHigh, "calculated high does not match expected high for percentile %v", percentile)
+			}
+
+			assertOptionalFloat(t, "min", tc.Expected.Min, tc.Input.Min)
+			assertOptionalFloat(t, "max", tc.Expected.Max, tc.Input.Max)
+			assert.InDelta(t, tc.Expected.Average, tc.Input.Sum/float64(tc.Input.Count), 0.01)
```

Comment:
is delta calc conservative or more releaxed? just asking since we have seen some flakiness with tight delta calcs before